rust/src/address.rs

```diff
@@ -307,39 +307,45 @@ impl JsonSchema for Address {
 // other CBOR types
 #[wasm_bindgen]
 impl Address {
+    pub fn header(&self) -> u8 {
+        match &self.0 {
+            AddrType::Base(base) => ((base.payment.kind() as u8) << 4)
+                           | ((base.stake.kind() as u8) << 5)
+                           | (base.network & 0xF),
+            AddrType::Ptr(ptr) => 0b0100_0000
+                               | ((ptr.payment.kind() as u8) << 4)
+                               | (ptr.network & 0xF),
+            AddrType::Enterprise(enterprise) => 0b0110_0000
+                               | ((enterprise.payment.kind() as u8) << 4)
+                               | (enterprise.network & 0xF),
+            AddrType::Reward(reward) => 0b1110_0000
+                                | ((reward.payment.kind() as u8) << 4)
+                                | (reward.network & 0xF),
+            AddrType::Byron(_) => 0b1000,
```

Comment:
0b1000 are bits 7-4 so this should at least be 0b1000_0000. The explanation in `alonzo.cddl` doesn't mention network ID for byron addresses but they're the lower bits in other address types. Can you not use them outside of mainnet or was that just an omission in that header documentation? Do they map the old protocol magic to Shelley+ equivalents?

```
; shelley payment addresses:
; bit 7: 0
; bit 6: base/other
; bit 5: pointer/enterprise [for base: stake cred is keyhash/scripthash]
; bit 4: payment cred is keyhash/scripthash
; bits 3-0: network id
;
; reward addresses:
; bits 7-5: 111
; bit 4: credential is keyhash/scripthash
; bits 3-0: network id
;
; byron addresses:
; bits 7-4: 1000

; 0000: base address: keyhash28,keyhash28
; 0001: base address: scripthash28,keyhash28
; 0010: base address: keyhash28,scripthash28
; 0011: base address: scripthash28,scripthash28
; 0100: pointer address: keyhash28, 3 variable length uint
; 0101: pointer address: scripthash28, 3 variable length uint
; 0110: enterprise address: keyhash28
; 0111: enterprise address: scripthash28
; 1000: byron address
; 1110: reward account: keyhash28
; 1111: reward account: scripthash28
; 1001 - 1101: future formats
```